### PR TITLE
Separate linux and mac build and test jobs

### PR DESCRIPTION
**Pull request by mortalisk**
This allows linux tests to start without waiting for mac wheels to build.

The build and test jobs have been extacted into reusable workflows, which are included from the new build-mac, build-linux, test-linux and test-max jobs.

The aim was to also use xdist for this, but it requires some more work. That will be done in anther PR.

With xdist these changes give feedback for the ubuntu runners after ~8 minutes as compared to ~28 minutes. As is now, it takes ~14 minutes.

### CODE REVIEW

File: .github/workflows/test_ert.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      os:
+        type: string
+      python-version:
+        type: string
+      test-type:
+        type: string
+jobs:
+
+  tests-ert:
+    name: Run ert tests yo
```
Comment (eivindjahren) [inline .github/workflows/test_ert.yml:13]: Don't be cute.

File: .gitignore [hunk 2]
```diff
@@ -35,3 +35,4 @@ src/ert/shared/version.py
 /compile_commands.json
 *_pb2.py
 *_pb2.pyi
+/wheelhouse
```
Comment (eivindjahren) [inline .gitignore:38]: wai do i need diz?

File: .github/workflows/test_ert.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    - name: Test for a clean repository
+      run: |
+        # Run this before the 'Test CLI' entry below, which produces a few files that are accepted for now. Exclude the wheel.
+        git status --porcelain | sed '/ert.*.whl$/d'
+        test -z "$(git status --porcelain | sed '/ert.*.whl$/d')"
+
+    - name: Test CLI
+      run: |
+        ert --help
```
Comment (eivindjahren) [inline .github/workflows/test_ert.yml:63]: Missing new line here.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -247,14 +188,14 @@ jobs:
         mv setup.cfg tmp_tests/setup.cfg
         mkdir tmp_tests/.git
 
-    - name: Test docs
+    - name: Test docses
```
Comment (eivindjahren) [inline .github/workflows/build.yml:191]: Uh

Comment (mortalisk) [reply .github/workflows/build.yml:191]: don't even know :open_mouth:

**Review by eivindjahren — APPROVED** [review]

**Pull request merged**